### PR TITLE
Consolidate static classes

### PR DESCRIPTION
**Pull request by mxashlynn**
At your suggestion I went ahead and replaced the AllParquets and AllItems classes with properties in a central class, which I named "All".  I moved other game-specific data here from AssemblyInfo as well.

To support this, I had to add a layer of complexity to EntityCollection as AllParquets does some type checking and bounds checking that the other EntityCollections do not do.

Because this refactored AllParquets, many files were touched; however, the substantial changes are confined to:
- EntityCollection
- AssemblyInfo
- All

### CODE REVIEW

File: ParquetClassLibrary/All.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        #endregion
+
+        #region Collections
+        /// <summary>
+        /// A collection of all defined parquets of all subtypes.
+        /// This collection is the source of truth about parquets for the rest of the library,
+        /// something like a color palette that other classes can paint with.
+        /// </summary>
+        /// <remarks>All <see cref="EntityID"/>s must be unique.</remarks>
+        public static EntityCollection<ParquetParent> Parquets { get; set; }
```
Comment (WispyMouse) [inline ParquetClassLibrary/All.cs]: Are collections intentionally not marked with Readonly?

File: ParquetClassLibrary/EntityCollection.cs [hunk 2]
```diff
@@ -106,29 +142,34 @@ public bool Remove(Entity in_entity)
...
-        /// <param name="in_id">A valid, defined <see cref="Entity"/> identifier.</param>
-        /// <typeparam name="T">The type of <see cref="Entity"/> sought.  Must correspond to the given ID.</typeparam>
-        /// <returns>The specified <see cref="Entity"/>.</returns>
-        public Entity Get(EntityID in_id)
+        /// <param name="in_id">A valid, defined <typeparamref name="T"/> identifier.</param>
+        /// <typeparam name="T">
+        /// The type of <typeparamref name="ParentType"/> sought.  Must correspond to the given <paramref name="in_id"/>.
+        /// </typeparam>
+        /// <returns>The specified <typeparamref name="T"/>.</returns>
+        public T Get<T>(EntityID in_id) where T : ParentType
```
Comment (WispyMouse) [inline ParquetClassLibrary/EntityCollection.cs:161]: comment: Elegant solution. I was going to ask, can an EntityCollection<Critter> return something other than a Critter? Why not just have Get always return a Critter? Then I considered a child class of Critter, and saw that where T : ParentType clause, making this a nice future proof solution.

File: ParquetClassLibrary/EntityCollection.cs [hunk 3]
```diff
@@ -106,29 +142,34 @@ public bool Remove(Entity in_entity)
...
-        /// </summary>
-        /// <returns>An enumerator that iterates through the collection.</returns>
-        public IEnumerator<Entity> GetEnumerator()
-        {
-            return Entities.Values.GetEnumerator();
+            return (T)Entities[in_id];
         }
+        #endregion
 
+        #region Utilitie Methods
```
Comment (WispyMouse) [inline ParquetClassLibrary/EntityCollection.cs]: nit: Is it Utility or Utilitie? May be a British spelling.

File: ParquetClassLibrary/All.cs (hunk 1 above)
Comment (mxashlynn) [reply ParquetClassLibrary/All.cs]: Let me double-check~

File: ParquetClassLibrary/EntityCollection.cs (hunk 2 above)
Comment (mxashlynn) [reply ParquetClassLibrary/EntityCollection.cs:161]: Thank you!  ^///^

File: ParquetClassLibrary/EntityCollection.cs (hunk 3 above)
Comment (mxashlynn) [reply ParquetClassLibrary/EntityCollection.cs]: oops, that was supposed to be "Utilities"  XD

File: ParquetClassLibrary/All.cs (hunk 1 above)
Comment (mxashlynn) [reply ParquetClassLibrary/All.cs]: That was in error.  I must have had the old pattern from before the refactor in my head.  Fixed!

**Pull request merged**